src/core/Darwin.zig

```diff
@@ -76,193 +59,299 @@ pub fn run(comptime on_each_update_fn: anytype, args_tuple: std.meta.ArgsTuple(@
     // TODO: support UIKit.
 }
 
-pub fn init(
-    darwin: *Darwin,
+pub fn tick(core: *Core) !void {
+    var windows = core.windows.slice();
+    while (windows.next()) |window_id| {
+        if (core.windows.getAll(window_id)) |cw| {
+            const core_window = cw;
+            const native: Native = core_window.native;
+
+            if (core.windows.get(window_id, .device).? == null) {
+                // Window has not been set up yet, we need to init the window
+                //try initWindow(core, window_id);
+                //try core.initWindow(window_id);
+
+                try initWindow(core, window_id);
+                try core.initWindow(window_id);
+            }
+            if (native.window) |native_window| {
+                // Handle resizing the window when the user changes width or height
+                if (core.windows.updated(window_id, .width) or core.windows.updated(window_id, .height)) {
+                    var frame = native_window.frame();
+                    frame.size.height = @floatFromInt(core.windows.get(window_id, .width).?);
+                    frame.size.width = @floatFromInt(core.windows.get(window_id, .height).?);
+                    native_window.setFrame_display_animate(frame, true, true);
+                }
+            }
+        }
+    }
+}
+
+fn initWindow(
     core: *Core,
-    options: InitOptions,
+    window_id: mach.ObjectID,
 ) !void {
-    var surface_descriptor = gpu.Surface.Descriptor{};
-
-    // TODO: support UIKit.
-    var window_opt: ?*objc.app_kit.Window = null;
-    if (!options.headless) {
+    if (core.windows.getAll(window_id)) |cw| {
```

Comment:
Oh! I also think that `getAll` cannot return a `null` value ever, we should fix that too